tests/integ/scala/test_dataframe_writer_suite.py

```diff
@@ -139,6 +134,73 @@ def test_negative_write_with_target_column_name_order(session):
         session.table(table_name).drop_table()
 
 
+def test_write_with_target_column_name_order_all_kinds_of_dataframes_without_truncates(
+    session,
+):
+    table_name = Utils.random_table_name()
+
+    session.create_dataframe(
+        [],
+        schema=StructType(
+            [StructField("a", IntegerType()), StructField("b", IntegerType())]
+        ),
+    ).write.save_as_table(table_name, table_type="temporary")
+
+    try:
+        large_df = session.create_dataframe([[1, 2]] * 1024, schema=["b", "a"])
+        large_df.write.save_as_table(
+            table_name, mode="append", column_order="name", table_type="temp"
+        )
+        rows = session.table(table_name).collect()
+        assert len(rows) == 1024
+        for row in rows:
+            assert row["B"] == 1 and row["A"] == 2
+    finally:
+        session.table(table_name).drop_table()
+
+
+def test_write_with_target_column_name_order_with_nullable_column(
```

Comment:
this test in live tests failed because of mismatching column: https://github.com/snowflakedb/snowpark-python/actions/runs/9846922564/job/27185862288?pr=1872

```
E   snowflake.snowpark.exceptions.SnowparkSQLException: (1304): 01b589f8-090a-a60b-000c-a904df91c6e7: 002020 (21S01): SQL compilation error:
E   Insert value list does not match column list expecting 6 but got 2
```